### PR TITLE
Create webhook which if hit will fetch the latest feature switches from Bolt

### PR DESCRIPTION
**Pull request by vitaliyreznikov**
Fixes: https://app.asana.com/0/1163213880723410/1163214273078115/f

#changelog Create webhook which if hit will fetch the latest feature switches from Bolt

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Test

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: tests/unit/testsuite/Bolt/Boltpay/controllers/FeatureswitchesControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * Setup test dependencies, called before each test
+     *
+     * @throws Mage_Core_Exception if unable to stub helper
+     */
+    protected function setUp()
+    {
+        $this->currentMock = $this->getTestClassPrototype()
+            //->setMethods(array('getResponse', 'sendResponse', 'getRequestData'))
```
Comment (adentong) [inline tests/unit/testsuite/Bolt/Boltpay/controllers/FeatureswitchesControllerTest.php]: Remove this line.

File: app/code/community/Bolt/Boltpay/controllers/FeatureswitchesController.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+     * Send to bolt API request to update feature switches
+     * Send response with status - success or failure
+     */
+    public function changedAction()
+    {
+        $updateResult = Mage::getSingleton("boltpay/featureSwitch")->updateFeatureSwitches();
+        if ($updateResult) {
+            $this->sendResponse(self::HTTP_OK, array('status' => 'success'));
+        } else {
+            $this->sendResponse(self::HTTP_UNPROCESSABLE_ENTITY, array('status' => 'failure'));
```
Comment (adentong) [inline app/code/community/Bolt/Boltpay/controllers/FeatureswitchesController.php]: Curious why 422? Is there another status we could use that's better?

Comment (vitaliyreznikov) [reply app/code/community/Bolt/Boltpay/controllers/FeatureswitchesController.php]: We use code 422 everywhere in the plugin and it looks like it also good for this situation. From RFC:
---
422 UNPROCESSABLE ENTITY
The server understands the content type of the request entity (hence a 415 Unsupported Media Type status code is inappropriate), and the syntax of the request entity is correct (thus a 400 Bad Request status code is inappropriate) but was unable to process the contained instructions.
---
The request contains the instruction "get feature switches from server" and we are unable to process this instruction for some reason.
But I am open to any suggestions.

Comment (adentong) [reply app/code/community/Bolt/Boltpay/controllers/FeatureswitchesController.php]: Got it. I was just a little curious.

File: app/code/community/Bolt/Boltpay/controllers/FeatureswitchesController.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    extends Mage_Core_Controller_Front_Action implements Bolt_Boltpay_Controller_Interface
+{
+    use Bolt_Boltpay_Controller_Traits_WebHookTrait;
+
+    /**
+     * Initializes Controller member variables
+     */
+    public function _construct()
+    {
+        //TODO: remove this method after adding signature support on bolt side
```
[the remark this answers is not in the document]
Comment (roopakv) [reply app/code/community/Bolt/Boltpay/controllers/FeatureswitchesController.php]: yeah i should remove this todo. i initially wanted to secure it, but decided it didn't matter.

here is my thought process.
- the endpoint does not accept any input whatsoever
- this endpoint does not do anything but trigger calling bolt to get the feature switches.
- the worse thing someone can do is DDOS the merchant server, but fi someone wanted to do this, the signature check does not prevent this, and there are so many more expensive calls that are bigger targets.

given all of this, I think this endpoint can be unsecured. It is simpley a mechanism to tell the plugin to go get the latest switches from bolt.

LMK if you disagree.

[the remark this answers is not in the document]
Comment (mccottry) [reply app/code/community/Bolt/Boltpay/controllers/FeatureswitchesController.php]: I agree with you.

**Pull request merged**